### DISCUSSION TITLE
General allometric functions in FATES

### DISCUSSION THREAD

**Discussion by mpaiao** [NGEET/fates#1123] [Ideas]
As part of defining multiple tropical PFTs for FATES, I am also fitting and comparing multiple allometric models based on observations on multiple structural properties (height, aboveground biomass, crown area, crown depth, leaf biomass/leaf area, bark thickness), using existing models in the literature. 

 In many cases the selected model formulation doesn't exist in FATES, and I am intended to implement them. My only concern is that this may end up adding too many allometric modes that are quite similar to the existing ones. To give an example, for aboveground biomass, the selected model has the form:
```
    agb = a1 * (dbh^2 * height)^a2 * wood_density^a3
```

which is similar to the existing `dh2bagw_salda` and `dh2bagw_chave2014` (and arguably d2bagw_2pwr too). But mathematically, we could define a general allometric equation:

```
    agb = a1 * dbh^a2 * height^a3 * wood_density^a4
```

that can be reduced to the model I fitted and the three existing modes, by setting some of the parameters to zero, or by setting `a2 = 2 * a3`, `a4 = a3`, and so on in the parameter file. I think the general equation would make model fitting more flexible, but the link to published allometric equations (like Saldarriaga et al., Chave et al.) might become less transparent and may also require refactoring many default parameters in FATES. 

I am happy to implement the model either way, but I thought I should check and see what people's thoughts and preferences are.

**Comment by glemieux** [comment]
I like the cleanliness of the general equation method, but I can see your point about it obfuscating the link to the source equation.  Although, its a minor issue, I suspect the generalized method would be _slightly_ more prone to error given that the developer would need to convert the specific equation to a generalized form.  My assumption is that there is also no real performance impact with either method; I'm assuming that these modes won't be dynamic, but set at compile time.  As such, I don't have a strong feeling either way.

**Comment by mpaiao** [comment]
Thanks for the feedback! I think will implement the new formulations as new modes, so it is clearer for future users/developers.
